### PR TITLE
add update.always and update.require_automerge_label

### LINKED ISSUE #174: use update_branch_immediately=true with require_automerge_label=true
we use the following kodiak config

```
# .kodiak.toml
version = 1

[merge]
method = "squash"
# once a PR is merged into master, delete the branch
delete_branch_on_merge = true
# label to use to enable Kodiak to merge a PR
automerge_label = "automerge" # default: "automerge"
# require that the automerge label be set for Kodiak to merge a PR. if you
# disable this Kodiak will immediately attempt to merge every PR you create
require_automerge_label = true
# if a PR is passing all checks and is able to be merged, merge it without
# placing it in the queue. This will introduce some unfairness where those
# waiting in the queue the longest will not be served first.
prioritize_ready_to_merge = true
# immediately update a PR whenever the target updates. If enabled, Kodiak will
# not be able to efficiently update PRs. Any time the target of a PR updates,
# the PR will update.
update_branch_immediately = true

[merge.message]
title = "pull_request_title"
body = "github_default"
include_pr_number = true
strip_html_comments = true # remove html comments to auto remove PR templates
```

and we have the following use-case:
we would like kodiak to automatically merge changes from the target branch into the PR, but do not yet merge them. therefore we use a `WIP:` title-prefix and mark PRs as soon as we want kodiak to update them with the merge-label.

it seems kodiak does not start auto-merging in changes from the target branch as long the title uses `WIP:` prefix... any chance we can get the desired behaviour?

### LINKED ISSUE #213: Clarifications about the workflow
We're having the problem that kodiak does not update the branches of our PRs when the base branch receives some changes:
image

It looks like kodiak is first waiting for the approvals, then will update the branch? However, none of the team members will ever give an approval to a branch which is not up to date. Therefore we end up manually updating our PRs.

We hoped that kodiak would first ensure the branch is up to date before performing any other check.

Is there a way to specify the order of the kodiak operations?

### PR DESCRIPTION
**Pull request by chdsbd**
CHANGES
- add config `update.always` to trigger Kodiak to update a branch regardless of Pr mergeability. 
- deprecate `merge.update_branch_immediately`. For Kodiak, a deprecated configuration option means its use is discouraged, not that it will be removed.
- correct code docs for `merge.block_on_reviews_requested` to indicate that it is deprecated.

fixes #174, fixes #213

### CODE REVIEW

File: kodiak/config.py [hunk 1]
```diff
@@ -72,6 +72,12 @@ class Merge(BaseModel):
     dont_wait_on_status_checks: List[str] = []
     # immediately update a PR whenever the target updates
     update_branch_immediately: bool = False
+    # ignore specific status checks that would normally prevent a PR from being
+    # updating. If for example the WIP status check was incomplete, we could
+    # specify "WIP" in this configuration to enable kodiak to update a branch
```
Comment (staabm) [inline kodiak/config.py]: Just to make sure we are on the same page: in our use case we dont use the WIP app from github marketplace, but the WIP prefix from `blacklist_title_regex` config

**Comment by chdsbd** [comment]
I've realized that to add this feature I need to rethink how Kodiak evaluates PRs.

My idea is to centralize all of the logic around setting GitHub statuses, updating PRs, and merging PRs into a pure function that accepts PR state and returns "actions" (e.g. update branch, set status context) to perform. I think this will improve the testability of Kodiak and remove complexity from other places in the app.

**Comment by staabm** [comment]
Thanks for the update. Sounds good :-)

File: kodiak/config.py [hunk 2]
```diff
@@ -57,6 +57,8 @@ class Merge(BaseModel):
     method: MergeMethod = MergeMethod.merge
     # delete branch when PR is merged
     delete_branch_on_merge: bool = False
+    # deprecated. this feature is flawed and cannot be fixed. see https://github.com/chdsbd/kodiak/issues/153#issuecomment-523057332.
```
Comment (staabm) [inline kodiak/config.py]: Maybe write DEPRECATED in upper case or on a separate line, so it will stand out?

File: kodiak/test_evaluation.py [hunk 3]
```diff
@@ -3334,3 +3334,257 @@ def test_get_merge_body_empty(pull_request: PullRequest) -> None:
...
+            reviews=[review],
+            contexts=[context],
+            check_runs=[check_run],
+            valid_signature=False,
+            valid_merge_methods=[MergeMethod.squash],
+            is_active_merge=False,
+            skippable_check_timeout=5,
+            api_call_retry_timeout=5,
+            api_call_retry_method_name=None,
+            #
```
Comment (sbdchd) [inline kodiak/test_evaluation.py]: nit

```suggestion
```

Comment (chdsbd) [reply kodiak/test_evaluation.py]: I'm using those `#`s to mark the arguments that are customized.

**Pull request merged**